File: Python Programs/plus_minus.py
def count_ratios(arr):
    # Count the total number of elements
    total_elements = len(arr)
    
    # Count the number of positive, negative, and zero elements
    positive_count = sum(1 for num in arr if num > 0)
    negative_count = sum(1 for num in arr if num < 0)
    zero_count = sum(1 for num in arr if num == 0)
    
    # Calculate the fractions
    positive_ratio = positive_count / total_elements
    negative_ratio = negative_count / total_elements
    zero_ratio = zero_count / total_elements
    
    # Print the fractions with three decimal places
    print("{:.3f}".format(positive_ratio))
    print("{:.3f}".format(negative_ratio))
    print("{:.3f}".format(zero_ratio))

# Take user input for the array
n = int(input())
arr = list(map(int, input().split()))

# Call the function with user input array
count_ratios(arr)
                                                                                                                            

File: Java Programs/plus_minus.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt(); // Number of elements in the array
        int[] arr = new int[n]; // Array to store the elements
        for (int i = 0; i < n; i++) {
            arr[i] = input.nextInt(); // Taking input for the array
        }
        calculateFractions(arr); // Calling method to calculate fractions
    }
    
    public static void calculateFractions(int[] arr) {
        int positiveCount = 0;
        int negativeCount = 0;
        int zeroCount = 0;
        int n = arr.length;
        
        // Counting positive, negative, and zero elements
        for (int i = 0; i < n; i++) {
            if (arr[i] > 0)
                positiveCount++;
            else if (arr[i] < 0)
                negativeCount++;
            else
                zeroCount++;
        }
        
        // Calculating fractions
        double positiveFraction = (double) positiveCount / n;
        double negativeFraction = (double) negativeCount / n;
        double zeroFraction = (double) zeroCount / n;
        
        // Printing fractions with three decimal places
        System.out.printf("%.3f%n", positiveFraction);
        System.out.printf("%.3f%n", negativeFraction);
        System.out.printf("%.3f%n", zeroFraction);
    }
}
                                                                                                                            